src/sagemaker/amazon/kmeans.py

```diff
@@ -33,8 +33,8 @@ class KMeans(AmazonAlgorithmEstimatorBase):
     epochs = hp('epochs', (gt(0), isint), 'An integer greater-than 0')
     center_factor = hp('extra_center_factor', (gt(0), isint), 'An integer greater-than 0')
 
-    def __init__(self, role, train_instance_count, train_instance_type, k, init_method=None,
-                 max_iterations=None, tol=None, num_trials=None, local_init_method=None,
+    def __init__(self, role, train_instance_count, train_instance_type, k, default_mini_batch_size=5000,
```

Comment:
Introducing another argument with default value is a little dangerous. If someone is calling this function will arguments set positionally, then this will fail.

That said, I think we can take this risk - the chance of this problem occuring is low and we're doing a new release. I'd recommend including constructor signature changes in the changelog.

Reply:
This is a required parameter and must be specified by the user. An alternative to constructor declaration will be fit enforcement. Let me rewrite it in that alternative form.